src/main/java/com/mongodb/hibernate/jdbc/MongoConnectionProvider.java

```diff
@@ -0,0 +1,126 @@
+ *
+ * <p>The following Hibernate JDBC properties will be relied upon by Hibernate's {@link Configurable} SPI mechanism:
+ *
+ * <ul>
+ *   <li>{@linkplain org.hibernate.cfg.JdbcSettings#JAKARTA_JDBC_URL jakarta.persistence.jdbc.url}
+ *   <li>{@linkplain org.hibernate.cfg.JdbcSettings#JAKARTA_JDBC_USER jakarta.persistence.jdbc.user}
+ *   <li>{@linkplain org.hibernate.cfg.JdbcSettings#JAKARTA_JDBC_PASSWORD jakarta.persistence.jdbc.password}
+ * </ul>
+ *
+ * <p>{@value org.hibernate.cfg.JdbcSettings#JAKARTA_JDBC_URL} property is mandatory and it maps to MongoDB's <a
+ * href="https://www.mongodb.com/docs/manual/reference/connection-string/">connection string</a>, in which database name
+ * must be provided to align with JDBC URL's convention. The other two JDBC properties are optional.
+ *
+ * @see ConnectionProvider
+ * @see JdbcSettings#JAKARTA_JDBC_URL
+ * @see JdbcSettings#JAKARTA_JDBC_USER
+ * @see JdbcSettings#JAKARTA_JDBC_PASSWORD
+ * @see <a href="https://www.mongodb.com/docs/manual/reference/connection-string/">connection string</a>
+ */
+public class MongoConnectionProvider implements ConnectionProvider, Configurable, Stoppable {
+
+    private @Nullable MongoClient mongoClient;
+
+    @Override
+    public Connection getConnection() {
+        throw new NotYetImplementedException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-29");
+    }
+
+    @Override
+    public void closeConnection(Connection connection) {
+        throw new NotYetImplementedException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-29");
+    }
+
+    @Override
+    public boolean supportsAggressiveRelease() {
+        return false;
+    }
+
+    @Override
+    public boolean isUnwrappableAs(Class<?> unwrapType) {
+        return false;
+    }
+
+    @Override
+    public <T> T unwrap(Class<T> unwrapType) {
+        throw new UnknownUnwrapTypeException(unwrapType);
+    }
+
+    @Override
+    public void configure(Map<String, Object> configValues) {
+        var jdbcUrl = configValues.get(JAKARTA_JDBC_URL);
+        if (jdbcUrl == null) {
+            throw new HibernateException("Configuration is required: " + JAKARTA_JDBC_URL);
+        }
+        if (!(jdbcUrl instanceof String)) {
+            throw new HibernateException(
+                    String.format("Configuration ('%s') value ('%s') not of string type", JAKARTA_JDBC_URL, jdbcUrl));
+        }
+        ConnectionString connectionString;
+        try {
+            connectionString = new ConnectionString((String) jdbcUrl);
+        } catch (IllegalArgumentException iae) {
+            throw new HibernateException("Invalid MongoDB connection string: " + jdbcUrl, iae);
+        }
+        var database = connectionString.getDatabase();
+        if (database == null) {
+            throw new HibernateException("Database must be provided in connection string: " + jdbcUrl);
```

Comment:
Just specifying the value of the `JAKARTA_JDBC_URL` property is not enough, we also need to inform users of the name of the property whose value we have an issue with.

Reply:
accepted